contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -0,0 +1,269 @@
+    /// https://www.notion.so/originprotocol/Limited-simplified-native-staking-accounting-67a217c8420d40678eb943b9da0ee77d
+    /// In short, after dividing by 32, if the ETH remaining on the contract falls between 0 and fuseIntervalStart,
+    /// the accounting function will treat that ETH as Beacon chain consensus rewards.
+    /// On the contrary, if after dividing by 32, the ETH remaining on the contract falls between fuseIntervalEnd and 32,
+    /// the accounting function will treat that as a validator slashing.
+    /// @notice Perform the accounting attributing beacon chain ETH to either full or partial withdrawals. Returns true when
+    /// accounting is valid and fuse isn't "blown". Returns false when fuse is blown.
+    /// @dev This function could in theory be permission-less but lets allow only the Registrator (Defender Action) to call it
+    /// for now.
+    /// @return accountingValid true if accounting was successful, false if fuse is blown
+    /* solhint-enable max-line-length */
+    function doAccounting()
+        external
+        onlyRegistrator
+        whenNotPaused
+        returns (bool accountingValid)
+    {
+        // pause the accounting on failure
+        accountingValid = _doAccounting(true);
+    }
+
+    // slither-disable-start reentrancy-eth
+    function _doAccounting(bool pauseOnFail)
+        internal
+        returns (bool accountingValid)
+    {
+        if (address(this).balance < consensusRewards) {
+            return _failAccounting(pauseOnFail);
+        }
+
+        // Calculate all the new ETH that has been swept to the contract since the last accounting
+        uint256 newSweptETH = address(this).balance - consensusRewards;
+        accountingValid = true;
+
+        // send the ETH that is from fully withdrawn validators to the Vault
+        if (newSweptETH >= MAX_STAKE) {
+            uint256 fullyWithdrawnValidators;
+            // safe since MAX_STAKE is hardcoded to 32ETH
+            unchecked {
+                // explicitly cast to uint256 as we want to round to a whole number of validators
+                fullyWithdrawnValidators = uint256(newSweptETH / MAX_STAKE);
+            }
+            activeDepositedValidators -= fullyWithdrawnValidators;
+
+            uint256 wethToVault = MAX_STAKE * fullyWithdrawnValidators;
+            IWETH9(WETH_TOKEN_ADDRESS).deposit{ value: wethToVault }();
+            // slither-disable-next-line unchecked-transfer
+            IWETH9(WETH_TOKEN_ADDRESS).transfer(VAULT_ADDRESS, wethToVault);
+            _wethWithdrawnToVault(wethToVault);
+
+            emit AccountingFullyWithdrawnValidator(
+                fullyWithdrawnValidators,
+                activeDepositedValidators,
+                wethToVault
+            );
+        }
+
+        uint256 ethRemaining = address(this).balance - consensusRewards;
+        // should be less than a whole validator stake
+        require(ethRemaining < 32 ether, "unexpected accounting");
```

Comment:
We seem to alternate between writing `32 ether` inline, or using MAX_STAKE, sometimes even inside the same function. We should probably use one consistently?

Reply:
I changed to use `FULL_STAKE`